octobot_tentacles_manager/base_worker/tentacle_worker.py

```diff
@@ -0,0 +1,199 @@
+#  Drakkar-Software OctoBot-Tentacles-Manager
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+from os import listdir, path
+from os.path import join
+from shutil import copyfile
+from asyncio import gather, Event, wait_for
+
+from octobot_commons.logging.logging_util import get_logger
+from octobot_tentacles_manager.configuration.global_tentacle_configuration import GlobalTentacleConfiguration
+from octobot_tentacles_manager.constants import TENTACLES_ARCHIVE_ROOT, TENTACLES_FOLDERS_ARCH, \
+    TENTACLE_CONFIG, CONFIG_SCHEMA_EXT, USER_TENTACLE_SPECIFIC_CONFIG_PATH, PYTHON_INIT_FILE, \
+    CONFIG_EXT, TENTACLES_REQUIREMENTS_INSTALL_TEMP_DIR, DEFAULT_TENTACLES_URL, DEFAULT_TENTACLE_CONFIG
+
+from octobot_tentacles_manager.util.file_util import find_or_create
+from octobot_tentacles_manager.util.tentacle_explorer import load_tentacle_with_metadata
+from octobot_tentacles_manager.util.tentacle_fetching import fetch_and_extract_tentacles
+
+
+class TentacleWorker:
```

Comment:
It seems that your `TentacleWorker` has methods for a unique tentacle setup and some for tentacle**s** setup. I think you should split into two classes : `TentacleWorker` and `TentaclesWorker`.